core/metrics-core-service/src/main/java/org/hawkular/metrics/core/jobs/JobsServiceImpl.java

```diff
@@ -71,7 +73,7 @@ public void shutdown() {
     @Override
     public Single<JobDetails> submitDeleteTenantJob(String tenantId) {
         return scheduler.scheduleJob(DeleteTenant.JOB_NAME, DeleteTenant.JOB_NAME, ImmutableMap.of("tenantId",
-                tenantId), new SingleExecutionTrigger.Builder().build());
+                tenantId), new SingleExecutionTrigger.Builder().withDelay(5, TimeUnit.MINUTES).build());
```

Comment:
The duration of the delay is somewhat arbitrary, but I used a delay to avoid lock contention. Let's say we try to schedule the job right away, which would be the next minute. There is a much greater chance of the job scheduler obtaining the time slice lock to start executing job for the next minute than for some point later on. And if that does happen, the job has to be resubmitted, resulting in more light weight transactions.